### PR TITLE
IKC-316 Manage topics (CRUD)

### PR DESCRIPTION
**Pull request by pbelke**

### CODE REVIEW

**Review by Pefes — CHANGES_REQUESTED** [review]

File: kouncil-frontend/libs/feat-topics/src/lib/topics/list/topics.component.ts [hunk 1]
```diff
@@ -171,4 +229,56 @@ export class TopicsComponent extends AbstractTableComponent implements OnInit, O
       })()
     };
   };
+
+  createTopic(topicName?: string) {
```
Comment (juris-97) [inline kouncil-frontend/libs/feat-topics/src/lib/topics/list/topics.component.ts:233]: Missing return type 'void'

File: kouncil-frontend/libs/feat-topics/src/lib/topics/list/topics.component.ts [hunk 2]
```diff
@@ -171,4 +229,56 @@ export class TopicsComponent extends AbstractTableComponent implements OnInit, O
...
+    let config = {
```
Comment (juris-97) [inline kouncil-frontend/libs/feat-topics/src/lib/topics/list/topics.component.ts:234]: This variable can be const

File: kouncil-frontend/libs/feat-topics/src/lib/topics/list/topics.component.ts [hunk 3]
```diff
@@ -171,4 +229,56 @@ export class TopicsComponent extends AbstractTableComponent implements OnInit, O
...
+      data: topicName,
+      width: '500px',
+      autoFocus: 'dialog',
+      panelClass: ['app-drawer']
+    };
+
+    let matDialogRef = this.dialog.open(TopicFormComponent, config);
```
Comment (juris-97) [inline kouncil-frontend/libs/feat-topics/src/lib/topics/list/topics.component.ts:241]: const

File: kouncil-frontend/libs/feat-topics/src/lib/topics/list/topics.component.ts [hunk 4]
```diff
@@ -171,4 +229,56 @@ export class TopicsComponent extends AbstractTableComponent implements OnInit, O
...
+
+    matDialogRef.afterClosed().subscribe(() => {
+      this.loadTopics();
+    })
```
Comment (juris-97) [inline kouncil-frontend/libs/feat-topics/src/lib/topics/list/topics.component.ts:245]: Missing ;

File: kouncil-frontend/libs/feat-topics/src/lib/topics/list/topics.component.ts [hunk 5]
```diff
@@ -171,4 +229,56 @@ export class TopicsComponent extends AbstractTableComponent implements OnInit, O
...
+        this.progressBarService.setProgress(true);
+        this.deleteTopic(topicName);
+      }
+    });
+  }
+
+  private deleteTopic(topicName: string) {
+    this.topicService.deleteSchema(topicName, this.servers.getSelectedServerId())
+    .pipe(first())
+    .subscribe(() => {
```
Comment (juris-97) [inline kouncil-frontend/libs/feat-topics/src/lib/topics/list/topics.component.ts:266]: deprecated usage, instead use: 
```
.subscribe({
      next: () => {

      },
      error: error => {

      }
    });
```

File: kouncil-frontend/libs/feat-topics/src/lib/topics/list/topics.demo.service.ts [hunk 6]
```diff
@@ -1,8 +1,8 @@
 import {Injectable} from '@angular/core';
-import {TopicsService} from './topics.service';
 import {from, Observable} from 'rxjs';
 import {demoTopics} from './topics.demo.data';
 import {Topics} from '@app/common-model';
+import {TopicsService} from "../topics.service";
```
Comment (juris-97) [inline kouncil-frontend/libs/feat-topics/src/lib/topics/list/topics.demo.service.ts:5]: Single quotes should be used

**Review by juris-97 — CHANGES_REQUESTED** [review]

File: kouncil-frontend/libs/feat-topics/src/lib/topic/topic-data.ts [hunk 7]
```diff
@@ -0,0 +1,6 @@
+export class TopicData {
+
+  name: string;
+  partitions?: number;
+  replicationFactor?: number;
```
[the remark this answers is not in the document]
Comment (pbelke) [reply kouncil-frontend/libs/feat-topics/src/lib/topic/topic-data.ts:5]: It is being used in the topic form component

**Review by juris-97 — APPROVED** [review]
Looks good!

**Review by juris-97 — APPROVED** [review]

**Pull request merged**